### DISCUSSION TITLE
Anyway to stop people from pasting Schematics with cheats?

### DISCUSSION THREAD

**Discussion by xavierlvll** [Pryaxis/TShock#2870] [Q&A] [answered]
Wondering if there is a way to stop this. either a permission im missing . or a plugin someone can point me to. 
Would like to achieve this without banning people sense its an anarchy server. But i don't think I could ban them if I tried, they can change there ip with a VPN and there uuid in a flash

**Comment by xavierlvll** [comment]
I also have this happen a lot https://cdn.discordapp.com/attachments/765799734086467584/1050507650670002186/image.png were a person will paste like key lock blocks. anyway to stop this too?

**Reply by punchready** [reply]
This specific thing should be blocked with the recent update to cheat protection. If you are running the latest tShock version and still experience people actively placing those, please open another ticket for that. - It can only be bypassed with the permission xtpkt mentioned below, make sure that you did not give this permission to everyone.

**Reply by xavierlvll** [reply]
thanks for letting me know

**Comment by sgkoishi** [comment]
It's not quite feasible to correctly distinguish between build from scratch and paste from somewhere with cheat. TShock has tile banning (specific tile) and placement speed restrictions (not too fast), tho they won't stop cheaters from pasting 'slowly'.

**Reply by sgkoishi** [reply]
By default, TShock perform basic checks like “if you are holding the matching items and created tiles” and “if you are placing tiles too far away”. But if they do hold matching tiles and just manually placing tiles, there's nothing smart enough to detect such behaviour automatically.

**Comment by xtpkt** [comment, marked as answer]
Perchance are they running around with `tshock.ignore.sendtilesquare`? First step would be to revoke that and any other related permissions if you have granted them.

Otherwise sounds like you need to write a plugin that hooks into OnTileEdit to try to ease this.

**Reply by xavierlvll** [reply]
i'll give it a shot! i don't remember adding the perm, but i also worked on the server at like 4 am so who knows lol. thanks for the help!
